DataDog/datadog-operator
# [features][admission controller] update rbac for mutating webhook

### What does this PR do?

Update RBAC for mutating webhook configuration resource such that most verbs apply to a specified resourceName only.

### Motivation

Improved security

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

N/A

### Describe your test plan

- Deploy the Operator
- Deploy a DatadogAgent resource with the admission controller enabled

```
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    credentials:
      apiKey: [...]
      appKey: [...]
    # need this for kind
    kubelet:
      tlsVerify: false
  features:
    admissionController:
      enabled: true
      mutateUnlabelled: true
```
- Run `k get clusterrole datadog-cluster-agent -oyaml` to check if RBAC properly specifies the `datadog-webhook` MutatingWebhookConfiguration resource
- Deploy an application pod and verify the admission controller injects the expected envvars, e.g. `DD_ENTITY_ID`
- Inspect the Cluster Agent and verify that `DD_ADMISSION_CONTROLLER_WEBHOOK_NAME` envvar is set

Repeat the steps above, but add a custom webhook name:

```
    admissionController:
      enabled: true
      mutateUnlabelled: true
      webhookName: "custom-webhook-name"
```

### Checklist
